src/main/java/bisq/core/arbitration/DisputeManager.java

```diff
@@ -549,232 +700,280 @@ public void onFault(String errorMessage) {
 
     // arbitrator receives that from trader who opens dispute
     private void onOpenNewDisputeMessage(OpenNewDisputeMessage openNewDisputeMessage) {
+        String errorMessage;
         Dispute dispute = openNewDisputeMessage.getDispute();
+        Contract contractFromOpener = dispute.getContract();
+        PubKeyRing peersPubKeyRing = dispute.isDisputeOpenerIsBuyer() ? contractFromOpener.getSellerPubKeyRing() : contractFromOpener.getBuyerPubKeyRing();
         if (isArbitrator(dispute)) {
             if (!disputes.contains(dispute)) {
                 final Optional<Dispute> storedDisputeOptional = findDispute(dispute.getTradeId(), dispute.getTraderId());
                 if (!storedDisputeOptional.isPresent()) {
-                    dispute.setStorage(getDisputeStorage());
+                    dispute.setStorage(disputeStorage);
                     disputes.add(dispute);
-                    sendPeerOpenedDisputeMessage(dispute);
+                    errorMessage = sendPeerOpenedDisputeMessage(dispute, contractFromOpener, peersPubKeyRing);
                 } else {
-                    log.warn("We got a dispute already open for that trade and trading peer.\n" +
-                            "TradeId = " + dispute.getTradeId());
+                    errorMessage = "We got a dispute already open for that trade and trading peer.\n" +
+                            "TradeId = " + dispute.getTradeId();
+                    log.warn(errorMessage);
                 }
             } else {
-                log.warn("We got a dispute msg what we have already stored. TradeId = " + dispute.getTradeId());
+                errorMessage = "We got a dispute msg what we have already stored. TradeId = " + dispute.getTradeId();
+                log.warn(errorMessage);
             }
         } else {
-            log.error("Trader received openNewDisputeMessage. That must never happen.");
+            errorMessage = "Trader received openNewDisputeMessage. That must never happen.";
+            log.error(errorMessage);
+        }
+
+        // We use the DisputeCommunicationMessage not the openNewDisputeMessage for the ACK
+        ObservableList<DisputeCommunicationMessage> messages = openNewDisputeMessage.getDispute().getDisputeCommunicationMessages();
+        if (!messages.isEmpty()) {
+            DisputeCommunicationMessage msg = messages.get(0);
```

Comment:
That is a bit hacky... Some messages cause the generation of a DisputeCommunicationMessage. The DisputeCommunicationMessages are the one we persist and use for checking the ack state. So thats why the DisputeCommunicationMessage is here used for setting the sourceUID instead of the actual msg.